internal/generate/olm-catalog/descriptor/descriptor_test.go

```diff
@@ -26,89 +26,154 @@ import (
 	"github.com/ghodss/yaml"
 	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/gengo/types"
 )
 
-const testFrameworkPackage = "github.com/operator-framework/operator-sdk/test/test-framework"
-
-func getTestFrameworkDir(t *testing.T) string {
-	t.Helper()
-	absPath, err := os.Getwd()
-	if err != nil {
-		t.Fatal(err)
-	}
-	sdkPath := absPath[:strings.Index(absPath, "internal")]
-	tfDir := filepath.Join(sdkPath, "test", "test-framework")
-	// parser.AddDirRecursive doesn't like absolute paths.
-	relPath, err := filepath.Rel(absPath, tfDir)
-	if err != nil {
-		t.Fatal(err)
-	}
-	return relPath
-}
+var (
+	relativeProjectRootDir       = filepath.Join("..", "..", "..", "..")
+	testDataRelativePathFromRoot = filepath.Join("internal", "generate", "testdata", "go")
+)
 
 func TestGetKindTypeForAPI(t *testing.T) {
-	cases := []struct {
+	multiAPIRootDir := filepath.Join("pkg", "apis")
+	singleAPIRootDir := filepath.Join("api")
+	group := "cache"
+	version := "v1alpha1"
+
+	subTests := []struct {
 		description string
-		pkg, kind   string
-		numPkgTypes int
-		wantNil     bool
+		// True if apis dir in the expected single or multi group layout
+		isExpectedLayout bool
```

Comment:
@hasbro17 move these 2 lines below the `wantNil bool`. This will fix the `maligned` error. https://paste.centos.org/view/017c27be
Having the `bool` between all of the `string`s causes a maligned struct in memory. https://medium.com/@sebassegros/golang-dealing-with-maligned-structs-9b77bacf4b97

Basically with your current layout it would look like this (X = 1 byte for the variable, P = padding)
```
string  16 XXXXXXXX
           XXXXXXXX
bool     1 X 
            PPPPPPP
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
int      4 XXXX
bool     1     X   
                PPP
```
By moving the bool to the bottom of the struct you get only 2 bytes of padding vs 10.
```
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
string  16 XXXXXXXX
           XXXXXXXX
int      4 XXXX
bool     1     X   
bool     1      X   
                 PP
```
I did this locally and the maligned error went away.

Reply:
@jmrodri Fascinating. Thanks for the fix and explanation.